Fix docker build failure

martgras opened the pull request:
Fix the docker build error from https://github.com/esphome/issues/issues/2041

## Description:

````docker run --rm -v c:\esphomedev\esphome-docs\:/data/esphomedocs -p 8000:8000 -it esphome/esphome-docs```` fails with 
````
Running Sphinx v1.8.4
loading translations [en]... done

Exception occurred:
  File "/usr/local/lib/python3.5/dist-packages/sphinx/registry.py", line 472, in load_extension
    mod = __import__(extname, None, None, ['setup'])
  File "/data/esphomedocs/schema_doc.py", line 26
    logger.info(f"{SCHEMA_PATH} not found. Not documenting schema.")
                                                                  ^
SyntaxError: invalid syntax
````

The reason for this error is python 3.5 used in xenial. 
Switching to ubuntu focal resolves the error 


**Related issue (if applicable):** fixes https://github.com/esphome/issues/issues/2041

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

martgras committed:
Fix docker build failure

Fix the docker build error from https://github.com/esphome/issues/issues/2041

frenck commented on Dockerfile:
```diff
@@ -1,4 +1,6 @@
-FROM ubuntu:xenial
+FROM ubuntu:focal
```
Instead, should use the official Docker container of Python, this makes it faster as well (especially if we used the Alpine versions instead).

martgras replied:
used python-slim now and removed python from apt commands
I didn't use the alpine image  because I found many recommendations against it . E.g. https://pythonspeed.com/articles/alpine-docker-python/

frenck replied:
That is because of the wheels, however, we have the wheels available for Alpine :)
(as Home Assistant builds them). But nice for a future update.

martgras replied:
to be honest I'm glad if I get python working at all and also not a docker pro  :)  So I'm pretty agnostic which images to use 

Should I rather use this for the PR ? 
A quick test didn't show a huge difference to the python-slim build but it is indeed pretty small 146MB
````
FROM python:alpine

ARG BUILD_ARCH
ENV WHEELS_LINKS=https://wheels.home-assistant.io/alpine-3.13/${BUILD_ARCH}/

RUN \
    apk add --no-cache \
    curl \
    git \
    make \
    doxygen \
    openssh-client 

RUN pip3 install --no-cache-dir --no-binary :all: \
    sphinx

EXPOSE 8000
WORKDIR /data/esphomedocs

CMD ["make", "webserver"]
````

frenck replied:
Nah let's stick with what we have for now, maybe in the future; We could add dev containers based on the current implementation as well (which isn't possible with Alpine).

martgras committed:
Switch from ubuntu to python-slim 

Using python:slim  instead of python reduces the image size from ~1.1 GB to 410MB

frenck reviewed (approved):
Thanks, @martgras 👍
